java/core/src/java/org/apache/orc/impl/RecordReaderImpl.java

```diff
@@ -695,8 +695,8 @@ static TruthValue evaluatePredicateProto(OrcProto.ColumnStatistics statsProto,
                    " include ORC-517. Writer version: {}",
           predicate.getColumnName(), writerVersion);
       return TruthValue.YES_NO_NULL;
-    } else if (category == TypeDescription.Category.DOUBLE ||
-        category == TypeDescription.Category.FLOAT) {
+    } else if ((category == TypeDescription.Category.DOUBLE ||
+        category == TypeDescription.Category.FLOAT) && cs instanceof DoubleColumnStatistics) {
```

Comment:
This doesn't match with the following goal of PR.
> This pr is aimed at adapting to read ORC files created by CUDF, which may have missing statistics in their DOUBLE/FLOAT columns.

Previously, we assume that FLOAT type also has `DoubleColumnStatistics`. Now, it breaks the assumption in case of `FLOAT` type with `FloatColumnStatistics`.

In order to match with the PR description, do we need a null check here?

Reply:
When specific statistics are missing, cs is a `ColumnStatistics` object.

Reply:
Oh, got it.